### ISSUE TITLE
Access violation in opencv_world310.dll when training on Windows 10/7

### ISSUE THREAD

**Issue by MrZander** [AlexeyAB/darknet#680]
When running the command `darknet.exe detector train data/obj.data yolo-obj.cfg darknet53.conv.74`
I get an exception:

> Unhandled exception at 0x00007FFB92111A2C (opencv_world310.dll) in darknet.exe: 0xC0000005: Access violation writing location 0x000000670000001B.

The application crashes, here is the last output:

> Loading weights from darknet53.conv.74...
>  seen 64
> Done!
> Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
>  If error occurs - run training with flag -dont_show
> Resizing
> 352

I have tried running with -dont_show and I get the same problem.

When debugging, the exception occurs on line 268 of http_stream.cpp:
```
	// HSV augmentation
	// CV_BGR2HSV, CV_RGB2HSV, CV_HSV2BGR, CV_HSV2RGB
	cv::Mat hsv_src;
	cvtColor(sized, hsv_src, CV_BGR2HSV);	// also BGR -> RGB
	
	std::vector<cv::Mat> hsv;
	cv::split(hsv_src, hsv); //Exception occurs here
```

Here is the call stack:
```
 	opencv_world310.dll!00007ffb92111a2c()
 	opencv_world310.dll!00007ffb921f7a15()
 	opencv_world310.dll!00007ffb921e755e()
 	opencv_world310.dll!00007ffb921e6a8a()
 	opencv_world310.dll!00007ffb92129413()
>	darknet.exe!image_data_augmentation(_IplImage * ipl, int w, int h, int pleft, int ptop, int swidth, int sheight, int flip, float jitter, float dhue, float dsat, float dexp) Line 268	
 	darknet.exe!load_data_detection(int n, char * * paths, int m, int w, int h, int boxes, int classes, float jitter, float hue, float saturation, float exposure, int small_object) Line 740
 	darknet.exe!load_thread(void * ptr) Line 827
 	pthreadVC2.dll!00007ffbcb8c69b0()	Unknown
 	[External Code]	
```

I have tried this on:
Windows 7 w/ GTX 1070
Windows 10 w/ GTX 970

Both using the same code base.
I tried a custom set of training data and the training data in the Yolo_Mark repository.  Same error on both.

Any suggestions on where to look next?

**Comment by AlexeyAB** [comment]
Hi,

* What OpenCV versin do you use?
* Did you compile `darknet.sln` for **x64 & Release**?
* What params did you change in your cfg-file?
* Can you show these params from your cfg file? 
https://github.com/AlexeyAB/darknet/blob/a2f3644e87c11c222f86aa2aeb0e31d858ebb024/cfg/yolov3.cfg#L785
https://github.com/AlexeyAB/darknet/blob/a2f3644e87c11c222f86aa2aeb0e31d858ebb024/cfg/yolov3.cfg#L698
https://github.com/AlexeyAB/darknet/blob/a2f3644e87c11c222f86aa2aeb0e31d858ebb024/cfg/yolov3.cfg#L612
https://github.com/AlexeyAB/darknet/blob/a2f3644e87c11c222f86aa2aeb0e31d858ebb024/cfg/yolov3.cfg#L14-L16

* Can you run training successfully without OpenCV? 
     `\darknet.sln` -> (right click on project) -> properties -> C/C++ -> Preprocessor -> Preprocessor Definitions, and remove here: `OPENCV;`

**Comment by mengkeat** [comment]
I've got a similar issue on the no_gpu version in debug mode. However in release x64,:
When compiled without OpenCV, I get "Cannot resize this type of layer: Not enough space". 
With OpenCV, "Cannot resize this type of layer: No error"

**Comment by AlexeyAB** [comment]
@mengkeat 

1. Do you see message `Resizing type ...` before `Cannot resize this type of layer`, can you show screenshot of this message?

2. Do you train Yolo v2 or Yolo v3 model?

3. Do you use this GitHub repository?

4. Can you train with `random=0` in the cfg-file?

**Comment by mengkeat** [comment]
@AlexeyAB 
1) Here's the screenshot of the message.
![yolo-screenshot](https://user-images.githubusercontent.com/1922608/39160987-07559e00-47a1-11e8-83cf-72a629f31daa.PNG)

2) I'm training Yolo V3. Basically using the data from https://timebutt.github.io/static/how-to-train-yolov2-to-detect-custom-objects/ but following YoloV3 instructions as per your github page.

3) I used the last release with hash d014f75 . With OpenCV 3.2. 

4) With random=0, error is the same. 
![yolo-random0](https://user-images.githubusercontent.com/1922608/39161081-729d985c-47a1-11e8-882a-c29a669271b1.PNG)

I might try changing OpenCV version to see if the error's still there. But nice work anyway !

**Comment by AlexeyAB** [comment]
@mengkeat 

1. Thanks! So problem in the [yolo] layer.

2. Can you show your command for training?

3. Try to use the latest version from this repository instead of release.


4. Did you change to `random=0` in the last `[yolo]` layer? https://github.com/AlexeyAB/darknet/blob/3df335bb50f890b12fa1a9965e91b0cf46d7902c/cfg/yolov3.cfg#L788

**Comment by mengkeat** [comment]
@AlexeyAB 

2) Command for training is: darknet_no_gpu.exe detector train data/nfpa-obj.data yolo-demo.cfg darknet53.conv.74

(3) I tried the latest repository version. But it seems to crash prematurely for some reason with message pop-up "The application was unable to start correctly (0xc0000022). Not sure what's the issue this time though. 

4) With release has d014f75 , setting random=0 works ! But with the recent repo version, I am experiencing the same issue as (3) .

**Comment by mengkeat** [comment]
@AlexeyAB 

I am not using CUDA / cuDNN.  All was done with the no gpu version.
